Lesson3

JustSenoko opened the pull request:
не уверена, в какой класс лучше было добавлять вызов методов работы с логом, выбрала MainWindow, хотя сильно сомневалась насчет Network

JustSenoko committed:
new package persistence
class UserRepository moved to package persistence
class User moved to package server

JustSenoko committed:
добавлено сохранение истории чата и ее загрузка при авторизации

JustSenoko committed:
исправлено количество загружаемых сообщений истории на 100

usharik commented on src/persistence/MessageRepository.java:
```diff
@@ -0,0 +1,78 @@
+
+        List<TextMessage> result = new ArrayList<>();
+
+        try (RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r")) {
```
На мой взгляд лучше было бы использовать BufferedReader. Код получился бы существенно проще. Также не уверен, что RandomAccessFile даст существенный выигрыш в производительности.

JustSenoko committed:
исправления
добавлен интерфейс работы с историей
addMsg сделан synchronised
методы парсинга истории перенесены в класс истории
